### ISSUE TITLE
Calendar: In Scan mode screen reader focus is going to calendar dates twice.

### ISSUE THREAD

**Issue by dzearing** [OfficeDev/office-ui-fabric-react#11140]
Tracking all of the issues brought up with the 03/15 accessibility pass on Calendar fabric control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B8cfe67c5-ce76-4083-b60e-4190dbd38206%7D&action=default&uid=%7B8CFE67C5-CE76-4083-B60E-4190DBD38206%7D&ListItemId=1872&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

**Comment by lorejoh12** [comment]
I dug into this one but don't really have a good fix. This seems like it's going to be pretty nasty to figure out, for not a significant gain, although it's definitely annoying for scan mode users. Scan mode itself doesn't work well in the control- arrowing left and right just reads "no character selected" as well.

investigation:
As far as I can tell from the DOM, the duplication issue stems from having a
```
<td>
    <button>
        <span>Data!</span>
    </button>
</td>
```
data structure for every cell. When scan mode reaches the td, it reads out all of the content inside of it- the data on the button (which has an aria label). You can verify this locally by adding a custom aria-label to the <td>, which Narrator will then choose to read instead. Then scan mode reaches the button, and re-reads the data inside of it (same aria label). The span has aria-hidden=true, so it gets ignored.

We can't apply a quick fix here by adding aria-hidden to the button (to prevent scan mode from reaching it), because then the aria-label won't be read when we reach the <td>. We also can't just make a blank aria label for the <td> because a) those get ignored, and b) then we'd have an empty cell for narrator, which makes even less sense than duplication. One possible solution is to move the aria-label out of the button and up to the <td>, and then apply aria-hidden to the button, but there's a bunch of other metadata that's currently being used for the button (disabled, selected, readonly) that would have to be figured out which increases likelihood of regression.

**Comment by khmakoto** [comment]
Like @lorejoh12 said, this is by design with how scan mode handles controls and text. From the Narrator team:

>This is by design from Scan mode (to stop on individual controls and text).
>
>Narrator tries to prioritize reading text content over element/control properties (if they provide the same information) as text provides more detailed information about content (text attributes).
>
>Which means that technically whenever Narrator reads a control, it should try to read the control and every label element. Depending on the content, this might make Narrator jump between different parts of the content surface.

I don't think there's an awful lot we could do here because of that constraint.
